src/gui/qt/gui.cpp

```diff
@@ -27,197 +27,286 @@ const int yOffset = -380;
 
 // Inspired by CEmu's approach to key events
 class KeyboardEventFilter : public QObject {
-    Q_OBJECT
+  Q_OBJECT
 
 public:
-    bool eventFilter(QObject* obj, QEvent* event) override;
+  bool eventFilter(QObject *obj, QEvent *event) override;
 };
 
-bool KeyboardEventFilter::eventFilter(QObject* obj, QEvent* event) {
-    if (event->type() == QEvent::KeyPress || event->type() == QEvent::KeyRelease) {
-        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
-        bool isPress = (event->type() == QEvent::KeyPress);
-        
-        switch (keyEvent->key()) {
-            case Qt::Key_Left:
-                setKeydown(38, isPress);
-                return true;
-            case Qt::Key_Right:
-                setKeydown(27, isPress);
-                return true;
-            case Qt::Key_Up:
-                setKeydown(28, isPress);
-                return true;
-            case Qt::Key_Down:
-                setKeydown(37, isPress);
-                return true;
-            case Qt::Key_X:
-                setKeydown(78, isPress);
-                return true;
-        }
+bool KeyboardEventFilter::eventFilter(QObject *obj, QEvent *event) {
+  if (event->type() == QEvent::KeyPress ||
+      event->type() == QEvent::KeyRelease) {
+    QKeyEvent *keyEvent = static_cast<QKeyEvent *>(event);
+    bool isPress = (event->type() == QEvent::KeyPress);
+
+    switch (keyEvent->key()) {
```

Comment:
This is good for now but in the future it will be better to have them assigned dynamically through the GUI and saved to a config file

Reply:
Clear. Need also to admit I didn't check the consistency of my proposal with what you already made for the SDL version.
By experience, mapping so many keys is always a pain. So dynamic assignment would be great in future version.